cdap-app-templates/cdap-etl/cdap-etl-batch/src/main/java/co/cask/cdap/etl/batch/mapreduce/ETLMapReduce.java

```diff
@@ -144,6 +144,7 @@ public void configure() {
     properties.put(Constants.Source.PLUGINID, pipelineIds.getSource());
     properties.put(Constants.Sink.PLUGINIDS, GSON.toJson(pipelineIds.getSinks()));
     properties.put(Constants.Transform.PLUGINIDS, GSON.toJson(pipelineIds.getTransforms()));
+    properties.put(Constants.Connections.PLUGINID, GSON.toJson(pipelineIds.getConnections()));
```

Comment:
I know you're following what was there before, but it seems simpler just to serialize the entire Pipeline rather than serialize each field individually.